wemake_python_styleguide/visitors/ast/statements.py

```diff
@@ -235,3 +254,67 @@ def visit_ClassDef(self, node: ast.ClassDef) -> None:
         all_args = [*node.bases, *[kw.value for kw in node.keywords]]
         self._check_indentation(node, all_args)
         self.generic_visit(node)
+
+
+@final
+@alias('visit_variable_defying_statement', (
+    'visit_With',
+    'visit_AsyncWith',
+    'visit_For',
+    'visit_AsyncFor',
+))
+class VariableUsedOutsideOfBlockVisitor(BaseNodeVisitor):
+    """
+    Responsible for detection of variable defined in `with` and `for` outside of it's body
+    """
+
+    def visit_variable_defying_statement(self, node: AnyVariableDefyingStatement) -> None:
+        """
+        Checks that variable is not used after statement body
```

Comment:
Misses `Raises` policy with the list of violation that is raised by this visitor.